### ISSUE TITLE
[Feature request] Open terminal in repository folder

### ISSUE THREAD

**Issue by Stuzer05** [GitFiend/Support#55]
Would be great to have a menu button to open terminal in a root of the repository in case you have to manually do something.

**Comment by GitFiend** [comment]
This would be nice to have. One thing that's caused me to put this off is that I don't know the users prefered terminal, though maybe this isn't a big deal.

Something you can do right now is Repo -> Open Repo in File Browser. You could then open your terminal from there. I'm considering making easier. E.g. Right-click on the background.

**Comment by Stuzer05** [comment]
@GitFiend I think it will be convenient to run a default terminal on windows and I'm sure there are ways to do it in Linux as well (`exo-open --launch TerminalEmulator` on XFCE desktop env)` it's important to add a setting to allow user to change the terminal. For example, to let specify the path to terminal and a command to execute.

Let %s be a path to the repo directory, so user could set his terminal as
`gnome-terminal -- bash -c "cd %s"`

This will allow any user on any system to setup the right terminal (if terminal supports initial commands, of course).

**Comment by GitFiend** [comment]
Yes, I agree with that train of logic. The problem is this now feels too complex and yuck to me. If I discover a more elegant way of implementing it I will consider it.

**Comment by Stuzer05** [comment]
@GitFiend Unfortunately, there is no way anyone can do a 100% working terminal detection system on every current and future OS. What I'm saying is that an auto-detection is a must yet manual setting cannot be left behind, as user should be able to change the default terminal (for example to ConEmu on windows). Or in case auto-detection didn't work correctly. There are many reasons manual terminal setting is a great solution alongside with auto-detection of the default terminal.
Manual setting could be a fast yet 100% working solution for this menu button feature.

**Comment by Stuzer05** [comment]
I see this way of working in nearly all git clients (which have this feature) and it's bothering many people (incl. me) some of them lack the feature of changing the default terminal application.
On windows, for example, I prefer to use ConEmu instead of the default cmd.

**Comment by GitFiend** [comment]
Yeah, when I use Windows I use Cmder. I would want GitFiend to split my current window rather than making a new tab or new window. Because of this, I wouldn't use this feature. I agree with the logic in what you are saying. I just don't want to spend time implementing features that I can't be fully happy with and I don't love. I would rather implement the missing feature that sent you to the terminal in the first place.

I found your comment about you and others being bothered by not having the option to set a default interesting. I love hearing more about problems and frustrations than I do about solutions. That's the sort of feedback that makes me reconsider and change my mind.

**Comment by Stuzer05** [comment]
Well, you cannot cover all problems which could be solved with an open terminal. I often need to quickly change some file, make a merge without committing `--no-commit` (would be nice to have some kind of checkbox btw). I want to say is that terminal is so essential, it's hard to oversell it's importance.

**Comment by GitFiend** [comment]
That's fair enough. I usually already have a terminal open for the repo I'm working in. For me, this is about focus. I want GitFiend to be only as complicated as it needs to be. I feel it's the job of your operating system to launch apps, not GitFiend.
